### PR TITLE
Feat/adp 1705 add stake pool mint to tx inespection util2

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

We must add new transaction inspectors for minting/burning, pool registration/retirements and  transaction metadata so this information can be easily retrieved from transactions.

# Proposed Solution

Add new inspectors for extracting minting/burning, registration/retirements and transaction metadata information.

# Important Changes Introduced

- The mintedAssetsInspector utility was added.
- The metadataInspector utility was added.
- The poolRegistrationInspector utility was added.
- The poolRetirementInspector utility was added.

### CODE REVIEW

**Review by mkazlauskas — APPROVED** [review]

**Review by mirceahasegan — APPROVED** [review]

File: packages/core/src/util/txInspector.ts [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+  Script,
+  ScriptType,
   StakeAddressCertificate,
   StakeDelegationCertificate,
   TokenMap,
   TxAlonzo,
   TxIn,
   Value
-} from '../Cardano';
+} from '../Cardano/types';
```
Comment (rhyslbw) [inline packages/core/src/util/txInspector.ts:22]:
```suggestion
} from '../Cardano';
```

File: packages/core/src/util/txInspector.ts [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
 import { BigIntMath } from '@cardano-sdk/util';
-import { coalesceValueQuantities, resolveInputValue, subtractValueQuantities } from '../Cardano/util';
-import { inputsWithAddresses, isAddressWithin } from '../Address/util';
-import { removeNegativesFromTokenMap } from '../Asset/util';
+import { assetNameFromAssetId, policyIdFromAssetId, removeNegativesFromTokenMap } from '../Asset/util';
+import { coalesceValueQuantities } from './coalesceValueQuantities';
+import { inputsWithAddresses, isAddressWithin } from '../Cardano/util/address';
```
Comment (rhyslbw) [inline packages/core/src/util/txInspector.ts:26]:
```suggestion
import { inputsWithAddresses, isAddressWithin, resolveInputValue } from '../Cardano/util';
```

File: packages/core/src/util/txInspector.ts [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
+import { nativeScriptPolicyId } from './nativeScript';
+import { resolveInputValue } from '../Cardano/util/resolveInputValue';
```
Comment (rhyslbw) [inline packages/core/src/util/txInspector.ts:28]:
```suggestion
```

File: packages/core/src/util/metadatum.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import * as fromCore from '../../CSL/coreToCsl';
-import * as toCore from '../../CSL/cslToCore';
-import { CSL } from '../..';
-import { Metadatum, MetadatumMap } from '../types';
+import { CSL } from '../CSL/CSL';
+import { Metadatum, MetadatumMap } from '../Cardano/types/AuxiliaryData';
```
Comment (rhyslbw) [inline packages/core/src/util/metadatum.ts:2]:
```suggestion
import { Metadatum, MetadatumMap } from '../Cardano';
```

**Review by rhyslbw — COMMENTED** [review]
Looks good @AngelCastilloB, nice work. I've highlighted a few imports that can be shortened (as highlighted by my IDE). Any reason we can't do this @mkazlauskas? It's all building locally :thinking:

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**